codegen/aws/core/src/main/java/software/amazon/smithy/python/aws/codegen/AwsUserAgentIntegration.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.smithy.python.aws.codegen;
+
+import java.util.Collections;
+import java.util.List;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolReference;
+import software.amazon.smithy.python.codegen.ConfigProperty;
+import software.amazon.smithy.python.codegen.integrations.PythonIntegration;
+import software.amazon.smithy.python.codegen.integrations.RuntimeClientPlugin;
+import software.amazon.smithy.utils.SmithyInternalApi;
+
+/**
+ * Adds a runtime plugin to set user agent.
+ */
+@SmithyInternalApi
+public class AwsUserAgentIntegration implements PythonIntegration {
+    @Override
+    public List<RuntimeClientPlugin> getClientPlugins() {
+        return List.of(
+                RuntimeClientPlugin.builder()
```

Comment:
A `RuntimeClientPlugin` is similarly collected to be used later, but this time the code to do so is [inside of smithy-python](https://github.com/smithy-lang/smithy-python/blob/095e1b92dee9821ed5356fdfb228b85cf59867f0/codegen/core/src/main/java/software/amazon/smithy/python/codegen/generators/ConfigGenerator.java#L37).  

I'll need to do a deeper dive here to understand how this works fully, but the gist is outlined in the comment below.